enforcer-parent/enforcer/src/main/java/org/wso2/choreo/connect/enforcer/api/RestAPI.java

```diff
@@ -284,12 +296,56 @@ private PolicyConfig genPolicyConfig(OperationPolicies operationPolicies) {
         return policyList;
     }
 
+    private MockedApiConfig getMockedApiOperationConfig(
+            org.wso2.choreo.connect.discovery.api.MockedApiConfig mockedApiConfig, String operationName) {
+        MockedApiConfig configData = new MockedApiConfig();
+        configData.setIn(mockedApiConfig.getIn());
+        configData.setName(mockedApiConfig.getName());
+        List<MockedResponseConfig> responses = new ArrayList<>();
+        if (mockedApiConfig.getResponsesList() != null) {
+            for (org.wso2.choreo.connect.discovery.api.MockedResponseConfig response :
+                    mockedApiConfig.getResponsesList()) {
+                MockedResponseConfig responseData = new MockedResponseConfig();
+                responseData.setCode(response.getCode());
+                responseData.setValue(response.getValue());
+                if (response.getHeadersList() != null) {
+                    List<MockedHeaderConfig> headers = new ArrayList<>();
+                    for (org.wso2.choreo.connect.discovery.api.MockedHeaderConfig header : response.getHeadersList()) {
+                        MockedHeaderConfig headerConfig = new MockedHeaderConfig();
+                        headerConfig.setName(header.getName());
+                        headerConfig.setValue(header.getValue());
+                        headers.add(headerConfig);
+                    }
+                    responseData.setHeaders(headers);
+                }
+                MockedPayloadConfig payload = new MockedPayloadConfig();
+                HashMap<String, String> payloadMap = new HashMap<>();
+
+                if (response.getPayload().getApplicationXML() == null &&
+                        response.getPayload().getApplicationJSON() == null) {
+                    logger.error("Mock API payloads not defined in the JSON script.");
+                } else {
+                    payloadMap.put(APIConstants.APPLICATION_XML, response.getPayload().getApplicationXML());
+                    payloadMap.put(APIConstants.APPLICATION_JSON, response.getPayload().getApplicationJSON());
+                }
+                payload.setPayloadMap(payloadMap);
+                responseData.setPayload(payload);
+                responses.add(responseData);
+            }
+        }
+        configData.setResponses(responses);
+        logger.debug("Mock API config processed successfully for the " + operationName + " operation.");
+        return configData;
+    }
+
     private void initFilters() {
-        // TODO : re-vist the logic with apim prototype implemetation
 
-        AuthFilter authFilter = new AuthFilter();
-        authFilter.init(apiConfig, null);
-        this.filters.add(authFilter);
+        // These filters will not be added if it's a mocked API
+        if (!apiConfig.isMockedApi()) {
```

Comment:
even mock implementation apis requires auth token in APIM. Shall we enable auth filters?

Reply:
checked on apim 4.1.0-m3